### ISSUE TITLE
[6] [diogogomes] UI Style Guide Implementation and Themes Improvements

### ISSUE THREAD

**Issue by diogogomes** [bitshares/bitshares-ui#1030]
*After working on issue #667, I have detected that it is possible to improve the entire UI Design and Development process.*

*@wmbutler has already been informed about this implementation by Telegram*

---

## UI Style Guide

Implementing a UI Style Guide helps keep the entire development team (designers or coders) on the same page.

In this way, there will be a flexible starting point to maintain the consistency of a visual style in BitShares Wallet that can also be applied to the web page and can also be worked with the marketing team.

This guide will help in the development of better user-oriented desktop and mobile user interfaces in an accessible way that users can understand.

---

## Components

- [x] Typography
- [x] Color Palette
- [x] Iconography
- [x] Forms
- [x] Alert modals
- [x] Breadcrumbs
- [x] Buttons
- [x] Cards
- [x] Checkboxes
- [x] Dialogs
- [x] Drop-down menus
- [x] Navigational menus (and subnavigation)
- [x] Radio buttons
- [x] Tables
- [x] Vertical lists
- [x] On-off switches

---

## Notes

- This UI Style Guide will be stored in a folder `docs/design` or in a repository `bidesy`. (*1)(*2)
- This UI Style Guide can start the Bidesy (BitShares Design System). (*2)
- Three themes will be taken into account.
- All design files will be stored in SVG being the standard more cross-software.
- All opinions from developers and users will be taken into consideration but the direction should be to adopt an easy-to-use interface, a clean and modern design.

##### *2. "Bidesy" it's just an abbreviation of "BitShares Design System" to make referencing more convenient.

**Comment by ahdigital** [comment]
Are you pursuing this @diogogomes as I can complete this task as a priority next week otherwise.

**Comment by diogogomes** [comment]
@ahdigital 
I'm in progress. I have not put anything here yet to not overflow this issue with information. I decided it would be best to review globally instead of each step.

@wmbutler , @xeroc what is your opinion about:

> - This UI Style Guide will be stored in a folder docs/design or in a repository bidesy. (*1)(*2)

**Comment by ahdigital** [comment]
My preference would be to have it located inside a docs/design folder. This way everything is together as one.

**Comment by wmbutler** [comment]
Inside the docs/design folder is best. @diogogomes you and @ahdigital should talk. Dominic has posted a lot of nice designs.

It's important that each of you come to a meeting of the minds regarding UX. We need cooperation here. Please review @ahdigital designs. He has put quite a bit of work into this and I think you will agree it looks great:

https://www.behance.net/gallery/62124671/BitShares-Browser-based-DEX

**Comment by diogogomes** [comment]
Hello @ahdigital 

I'm sorry for the time you spent in the Style Guide. We could have been coordinated from the beginning.

I sent you a message by Telegram on January 11 or 12 to invite you to work with me on this style guide but I haven't gotten any answer from you so far.

I agree with @wmbutler, we need cooperation.

In the face of our last conversation about design versioning, I know that you don't like to develop design using GitHub but as communication between us was not possible by Telegram, I suggest that we exchange ideas publicly here or on the [channel #development on Bitshares Discord](https://discord.gg/tpC4Z4d).

All suggestions are welcome!

**Comment by diogogomes** [comment]
Hello @wmbutler 

During this last month I was helping users on Discord and collecting opinions from non-users of Bitshares on other channels and forums.

After this research, the most important point that was transmitted is relatively to the very complex, sometimes confusing and intimidating interface.

I decided on a more friendly approach using the colors and font sizes in this way so that new users don't feel so intimidated.

I also received opinions that design is fine because DEX's target audience are professional traders. I do not agree with this point of view because if it were so we would have a normal volume of a professional exchange.


Thanks in advance for your feedback.


**This is PNG version. If you want, you can analyze with better quality in [this SVG version](https://gitlab.com/diogogomes.eu/bitshares-ui-style-guide-public/blob/master/BitShares%20Ui%20Style%20Guide.svg).**

![Bitshares Style Guide](https://gitlab.com/diogogomes.eu/bitshares-ui-style-guide-public/raw/master/BitShares%20UI%20Style%20Guide.png)

**Comment by wmbutler** [comment]
I'd like for you to work with @ahdigital on this. I think you both bring strengths to the table.

**Comment by diogogomes** [comment]
@wmbutler As I mentioned in the previous comment, I've been waiting for an answer from him since the day I talked to you, January 12th, even before I published this issue.

**Comment by SmileyChris** [comment]
I'm unsure about the colours of dialog buttons. If you needed several buttons, the guide doesn't make it clear which colours to use. Maybe "primary", "secondary", and "warning" colour states?

<sub>As a UX aside, buttons with "yes" and "no" are terrible - always use the actions themselves as the button options (e.g. "Proceed" / "Cancel").</sub>

**Comment by diogogomes** [comment]
Thanks for your input. About the yes/no buttons it is just an orientation for developers, not to implement like this.

About the needed of several buttons, it was told to me in one previous issue that the UI never should have more than two buttons.

About the colours, are you referring only to the dialog or all colours in general?

Thanks

**Comment by SmileyChris** [comment]
@diogogomes just the colours of the buttons, but independent of whether used in dialog boxes or elsewhere.

**Comment by ahdigital** [comment]
I must admit that I glanced over your Telegram message, probably because I was working on other issues for the last two sprints. I have however been engaged in the UX generally and now specifically this issue for the past few days.

Nevertheless, your update raises lots of questions for me so my combined feedback would be this:

## TL;DR
We need a much more detailed guide with practical examples going forwards. I feel like my comments will be taken negatively but that's not my intention at all. I'm just conscious that we need to be watertight for our developers and professional for our users and investors.

### Header
I actually like the idea of having the header of the light theme styled in blue, rather than the grey that I had in my example.

I think dark and midnight should be closer to the palette currently used though since this isn't a brand new product and we don't want to alienate our existing user base. I experimented with this in the [Behance](https://www.behance.net/gallery/62124671/BitShares-Browser-based-DEX)* link Bill posted above.

* This link was a prototype. I refined the styles in the guide that I started working on [most recently](https://www.behance.net/gallery/62549753/Bitshares-DEX-UI-style-guide).

### Typography

Did you make the sizes so big, particularly in the header because of your research?

> I decided on a more friendly approach using the colors and font sizes in this way so that new users don't feel so intimidated.

I don't feel these sizes work and that it's moved away from friendly and towards being childlike. I agree the UI needs to be simple but think about the image associated with traders - sleek, modern, cool, success, luxurious, money, classy, professional etc. Noto Sans Bold doesn't scream out these things to me.

Why are you introducing 2 font family's when Roboto Sans has all the weights we need and is the primary font used currently? The usage will get abused quickly and it's not really the developers fault if we don't explain the correct usage.

Despite this I don't actually see anywhere in your design where Roboto is used. The lightest weight used is in the footer and it's Noto Sans.

In the footer the 'Help' link is the warning colour. Will all hyperlinks be orange and therefore a warning? If not, then should a user be able to click the green 'Connected' text? Again, if not, how can the user differentiate.

### Iconograpy
Related the typography point above, I think we should move away from the Font Awesome icons because they are too 'bubbly' and big.

An icon pack with a thinner stroke would offer a cleaner appearance. For example, we're investigating Ant Design at the moment and these icons are [included as standard](https://ant.design/components/icon/#List-of-icons) which would fit well.

### Colours
We should split the guide into one for each theme to make it easier to implement well colour matched. components. Presently the colour palette looks like all colours can be used in all themes which I assume is not the case.

I see what you've done with the primary, warning, buy and sell colours but I think it makes the branding look too generic, like Bootstrap out of the box.

For me, some of the usage is confusing.  Why would the 'on' switch be blue and the primary action button on the dialog be green?

The zebra stripe colour of the table rows needs to be tweaked to avoid legibility issues when green text is used.

### Proportions

This may not have been intentional but the example UI looks strange in proportion on standard a resolution. Compare both our proposals against the existing:

![compare-ui](https://user-images.githubusercontent.com/3088454/36780164-0f127256-1c6a-11e8-9459-0ecf3eba2138.gif)

### Details
Without throwing more questions into the mix we need to be more detailed in these components:

- Typography (Sizes, use cases)
- Forms (An array of different inputs and states)
- Alert modals (Different types and styles)
- Breadcrumbs
- Buttons (Use cases)
- Dialogs
- Cards
- Tables

**Comment by diogogomes** [comment]
@SmileyChris 

Thanks for your suggestion but this style guide was not approved. It was @ahdigital instead. You can see it on [this link](https://www.behance.net/gallery/62549753/Bitshares-DEX-UI-style-guide?ilo0=1).

---

@ahdigital 

About the message was **before you started on Bitshares UI sprints**, I sent you when The day you were sharing your curriculum on the Telegram from Bitshares. I understand that you worked for a few days but I worked for a month on this (research and statistics included). The important thing is that you finally answered me.

>  I feel like my comments will be taken negatively but that's not my intention at all.
Of course not, I really like to hear opinions, particularly user opinions. The only issue for me was that I contacted you before you started and I spend so much time doing the research and the markdown docs alone taking this direction and I don't will be compensated fairly for my work...

@wmbutler is going to close this issue because it was decided it would be your Style Guide. **Your is [this](https://www.behance.net/gallery/62549753/Bitshares-DEX-UI-style-guide?ilo0=1), right?** In my opinion, you need to make one Style Guide cross theme and not only for one single theme. It's better to add more components and more colours too, it is missing a lot of colours and some components.

Even so, I would like to talk about the issues raised:

> We need a much more detailed guide with practical examples going forwards

I agree, I made this preview on SVG and the pull request was going to be a document in markdown with each separate component and the explanations for it, ie documentation implementing the Atomic Design for the designer/developers and of course not just this preview.

>  I'm just conscious that we need to be watertight for our developers and professional for our users and investors.

I agree, it's why I'm talking for mouths about one unified brand with the marketing group and one Design System for all Bitshares related in one independent repository and not only for the wallet. We've lack of coordination between bitshares.org and the wallet.

> dark and midnight should be closer to the palette currently used

My idea was to deliver one "new product" for all investors that are complaining about be tired of nothing new in the design. They want a "restart in the user interface and not just a few tweaks". It was their opinion, don't take negatively.

> Did you make the sizes so big, particularly in the header because of your research?

I just used +2px from the actual size on the header font. The header size is more small than the actual. Did you analysed properly my svg file?
![captura de ecra de 2018-02-28 17-47-00](https://user-images.githubusercontent.com/9022734/36803551-c210f7ac-1caf-11e8-8176-16e84d6cf6cd.png)


> I don't feel these sizes work and that it's moved away from friendly and towards being childlike. I agree the UI needs to be simple but think about the image associated with traders - sleek, modern, cool, success, luxurious, money, classy, professional etc. Noto Sans Bold doesn't scream out these things to me.

This approach never worked to bring volume to the exchange, one proof is our major rival growing their user base after "bring one user interface that every people on earth can use and not for advanced traders".

> Why are you introducing 2 font family's when Roboto Sans has all the weights we need and is the primary font used currently?

Because the greater part of DEX's volume comes from Asia and Roboto is _tofu_ and it can bring major issues rendering Asian characters. I choose Noto Sans because noto is the font _non tofu_ more similar to Roboto.

> Despite this I don't actually see anywhere in your design where Roboto is used. The lightest weight used is in the footer and it's Noto Sans.

Yes and with one propose, to drop Roboto for the previous reasons.

> In the footer the 'Help' link is the warning colour. Will all hyperlinks be orange and therefore a warning?Will all hyperlinks be orange and therefore a warning?

Like you can see the only warning link was the Help and I used a warning because the users never click on Help and are overflowing the chat rooms with basic questions. They don't know that we have an help on the DEX.

> If not, then should a user be able to click the green 'Connected' text? Again, if not, how can the user differentiate.

I don't understand this question. I made it like the actual version of the wallet. It will be changed in next versions?

> Related the typography point above, I think we should move away from the Font Awesome icons because they are too 'bubbly' and big.

It depends how are used, if you use it like a font are okay, if you use independent svgs, yes.
Any library are okay to is is used correctly and not like the actual implementation.

> We should split the guide into one for each theme to make it easier to implement well colour matched components. Presently the colour palette looks like all colours can be used in all themes which I assume is not the case.

I think in the opposite direction, get one common colour palette for all themes with the same base colours and only changing the background colours. It is more consistent for users and more easy for the development

> This may not have been intentional but the example UI looks strange in proportion on standard a resolution. Compare both our proposals against the existing:

I think you didn't review properly without zoom my svg file. My main navbar and sub navbar together are smaller in height that the actual implemented code, the only difference in the size are the fonts.

Once again, did you analized my svg file?

![captura de ecra de 2018-02-28 17-47-00](https://user-images.githubusercontent.com/9022734/36803440-713c14ce-1caf-11e8-895c-ab5fcd1a8640.png)

> Without throwing more questions into the mix we need to be more detailed in these components:

That was my idea from the beginning and the work I did in markdown.


**I created documentation in Markdown and I was to make a pull request but as I will not be paid properly for this work (I will only be paid 2 hours), I will keep private all the documentation I made until I am assigned a new task I that is paid properly.**

**Comment by wmbutler** [comment]
Crediting @diogogomes 2 hours on this and closing. Let's work together. Hopefully, you and @ahdigital can find common ground.
